src/storage/index.cpp

```diff
@@ -107,4 +119,47 @@ string Index::AppendRowError(DataChunk &input, idx_t index) {
 	return error;
 }
 
+//===--------------------------------------------------------------------===//
+// Stable storage functions
+//===--------------------------------------------------------------------===//
+
+void IndexStorage::WriteDummyBlockPointers(Serializer &serializer) {
+
+	vector<BlockPointer> index_pointers;
+	index_pointers.emplace_back(INVALID_BLOCK, 42);
+	serializer.WriteProperty(103, "index_pointers", index_pointers);
+}
+
+BlockPointer IndexStorage::GetBlockPointer(const IndexStorageInfo &info) {
+
+	D_ASSERT(info.properties.find("block_id") != info.properties.end());
+	D_ASSERT(info.properties.find("offset") != info.properties.end());
```

Comment:
Maybe instead of having key/value `properties` we can have inheritance, similar to what we have for e.g. `BindData` of functions, allowing us to use regular C++ properties while still allowing extension data to be serialized?

Reply:
I removed this from the PR altogether, as pluggable indexes are not necessarily the scope here. I'll think about inheritance when Max and I start adding the first pluggable index.